charts/etcd/templates/etcd-role.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Values.roleName }}
+  namespace: {{ .Release.Namespace }}
+  ownerReferences:
+  - apiVersion: druid.gardener.cloud/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Etcd
+    name: {{ .Values.name }}
+    uid: {{ .Values.uid }}
```

Comment:
I suggest to not add a `ownerReference` here because depending on the propagation policy owned resources could be deleted too early.
It's not guaranteed that the deletion of the `etcd` resource happens in a way that this role is only removed when the finalizer is patched out from the `etcd` resource.

Reply:
> It's not guaranteed that the deletion of the etcd resource happens in a way that this role is only removed when the finalizer is patched out from the etcd resource

@timuthy I didn't exactly get this. [Etcd deletion flow](https://github.com/gardener/etcd-druid/blob/0828abe1338b0654a33cb5b118a2d5eeeb9bcda5/controllers/etcd_controller.go#L280) first deletes the etcd statefulset and then removes the finalizers on the dependent secrets (backup secret and TLS secrets) and finally removes finalizer from the etcd resource. That will then ensure the deletion of the etcd and subsequently the etcd bootstrap configmap and service (as well as serviceaccount, role, rolebinding from this PR).

Reply:
>  That will then ensure the deletion of the etcd and subsequently the etcd bootstrap configmap and service (as well as serviceaccount, role, rolebinding from this PR).

Exactly this order you described depends on the `propagationPolicy`. Try to execute the following and you will see that all owned objects are gone immediately even though the Druid did not remove the finalizer from the `etcd` resource.

```
kubectl delete etcd etcd-test --cascade="foreground"
```

Reply:
In an out-of-band discussion we agreed with continuing with the `ownerReference` approach since we already took this for `ConfigMap` and `Service` resources and deleting the resources too early be accident is not very critical.